set.mm

```diff
@@ -466126,6 +466419,469 @@ C_ dom ( ( ( T ` { I , J } ) o. F ) \ _I ) ) $=
       XALJBUUBGKUUFWKVCWOWLWMXAUUIUUJWPWTKXQXECUUGDUUGWCWNWQWRWS $.
   $}
 
+  ${
+    $d D i $.  $d D w $.  $d I i $.  $d J i $.  $d M i $.  $d U i $.  $d W i $.
+    $d W w $.  $d i ph $.
+    cycpmco2.c $e |- M = ( toCyc ` D ) $.
+    cycpmco2.s $e |- S = ( SymGrp ` D ) $.
+    cycpmco2.d $e |- ( ph -> D e. V ) $.
+    cycpmco2.w $e |- ( ph -> W e. dom M ) $.
+    cycpmco2.i $e |- ( ph -> I e. ( D \ ran W ) ) $.
+    cycpmco2.j $e |- ( ph -> J e. ran W ) $.
+    cycpmco2.e $e |- E = ( ( `' W ` J ) + 1 ) $.
+    cycpmco2.1 $e |- U = ( W splice <. E , E , <" I "> >. ) $.
+    $( The composition of a cyclic permutation and a well-chosen transposition
+       is injective.  (Contributed by Thierry Arnoux, 4-Jan-2024.) $)
```

Comment:
I have a hard time understanding what's going on, but judging from the comment alone, isn't the composition of any two permutations injective? I.e. the transposition doesn't have to be well-chosen at all.

Reply:
Yes, I need to rewrite this comment. 

The cyclic permutation in question is not `U`, but ``( M ` U )``, and it's `U` which is injective.

`U` is actually the representation of a cycle. For example a word `W = <" A B C ">` would correspond to the cyclic permutation ``( M ` W )`` mapping `A` → `B`, `B` → `C`, `C` → `A`, `D` → `D`. 
In order for it to represent a cycle, the same element should not appear twice in the word, so it has to be injective.

Reply:
Maybe simply:
> The word `U` used in `~ cycpmco2` is injective, so it can represent a cycle and form a cyclic permutation ``( M ` U )``.

Reply:
I think that's better, yeah.